python/mister_webhooks/webhook_event.py

```diff
@@ -0,0 +1,111 @@
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum, auto
+import io
+import json
+from typing import Any, ByteString, Dict, List
+
+import avro.io
+import cbor2
+
+from .types import (
+    KafkaMessageEnvelopeV1,
+    KafkaMessageEnvelopeV1,
+    envelope_reader,
+)
+
+__all__ = ["WebhookEvent", "HTTPMethod"]
+
+
+class DecodeError(ValueError):
+    pass
+
+
+class EnvelopeDecodeError(DecodeError):
+    pass
+
+
+class PayloadDecodeError(DecodeError):
+    pass
+
+
+class HTTPMethod(Enum):
+    """HTTPMethod represents the HTTP method used to transmit the webhook event.
+
+    Only the document manipulation methods are encoded here, namely:
+
+    - GET
+    - HEAD
+    - POST
+    - PUT
+    - PATCH
+    - DELETE
+
+    """
+
+    GET = auto
+    HEAD = auto
+    POST = auto
+    PUT = auto
+    DELETE = auto
+    PATCH = auto
+
+
+@dataclass
+class WebhookEvent:
+    "WebhookEvent represents a received webhook event"
+
+    method: HTTPMethod
+    """HTTP method used to transmit the webhook event"""
+
+    timestamp: datetime
+    """Time at which the webhook event occurred"""
+
+    headers: Dict[str, List[bytes]]
+    """HTTP headers captured from the webhook request"""
+
+    payload: Any
+    """Webhook request body"""
+
```

Comment:
I hate to say it, but you write good python. Good thinking making this a dataclass.